awc/src/sender.rs

```diff
@@ -209,7 +209,8 @@ impl RequestSender {
     ) -> SendClientRequest {
         let body = match serde_json::to_string(value) {
             Ok(body) => body,
-            Err(e) => return Error::from(e).into(),
+            // TODO: own error type
+            Err(e) => return Error::from(io::Error::new(io::ErrorKind::Other, e)).into(),
```

Comment:
Thoughts on what do do here @fakeshadow, maybe a wrapped error that can also hold serialization errors as well as client errors from -http. Might leave it for a separate PR if you want to handle it.